### PR TITLE
fix: show issue overlay for processes on participant given collab.

### LINKED ISSUE #30: Display process lint errors on connected participant
__Is your feature request related to a problem? Please describe__

In collaboration diagrams we usually treat expanded participants like the process it is connected to, interaction wise. With https://github.com/bpmn-io/bpmn-js-bpmnlint/pull/29 merged we have the potential to better support this special case, too.

__Describe the solution you'd like__

* Display process lint errors not on the root but on the visible participant that represents it

__Describe alternatives you've considered__

* Keep existing behavior

__Additional context__

Current behavior (as of `v0.17.0`): Note that the process error is shown on the root, rather than the participant, which would be my expected position to display it.

capture QDWeit_optimized

### PR DESCRIPTION
**Pull request by MaxTru**
closes #30

Potential follow-up: the tests need a clean up. We should inject the issues rather than relying on bpmnlint to give certain issues.

### CODE REVIEW

**Comment by nikku** [comment]
Could you provide a small screen capture that shows the improved behavior in action? That helps to get feedback in quicker :)

**Comment by MaxTru** [comment]
Sure - good point.

Will follow-up with the too big pills

Screenshot_20210416_130015

**Comment by MaxTru** [comment]
cc @andreasgeier 
Screenshot_20210416_134056

**Comment by andreasgeier** [comment]
I assume it is exactly the same behavior as on the process level? So if the pool itself has errors, they would show up on top separated by a line? 

Then it's all good for me 👍

**Comment by MaxTru** [comment]
> I assume it is exactly the same behavior as on the process level? So if the pool itself has errors, they would show up on top separated by a line?
> 
> Then it's all good for me

Yep

File: lib/Linting.js [hunk 1]
```diff
@@ -154,15 +156,27 @@ Linting.prototype._formatIssues = function(issues) {
...
   reports = map(reports, function(report) {
     const element = self._elementRegistry.get(report.id);
 
     if (!element) {
       report.isChildIssue = true;
       report.actualElementId = report.id;
 
-      report.id = self._canvas.getRootElement().id;
+      // (2.1) Is a participant referring to the current issue? Then display on participant
+      const participants = self._elementRegistry.filter((ele) => { return is(ele, 'bpmn:Participant'); }),
```
Comment (nikku) [inline lib/Linting.js:168]: We could cache the participants / processes referenced by ID to speed things up (i.e. move them before the `#map`). Filtering across all diagram elements is a large operation we should not repeat unnecessarily.

File: lib/Linting.js [hunk 2]
```diff
@@ -154,15 +156,27 @@ Linting.prototype._formatIssues = function(issues) {
...
+            participantBos = participants.map((ele) => { return ele.businessObject; }),
+            referringParticipant = participantBos.filter((ele) => { return (ele.processRef && ele.processRef.id && ele.processRef.id === report.id); });
+
+      if (referringParticipant.length === 1) {
```
Comment (nikku) [inline lib/Linting.js:172]: Can it also be `2`? If not, let's simplify this to `referringParticipant.length`.

**Review by nikku — CHANGES_REQUESTED** [review]
Looks good. Two small comments attached.

File: lib/Linting.js (hunk 1 above)
Comment (MaxTru) [reply lib/Linting.js]: Godd point - done

File: lib/Linting.js (hunk 2 above)
Comment (MaxTru) [reply lib/Linting.js]: Good point - done

**Comment by MaxTru** [comment]
> Looks good. Two small comments attached.

Thanks for the review and comments incl. improvement suggestions! I updated the PR

**Review by nikku — APPROVED** [review]
Great work!

**Comment by nikku** [comment]
Released, marketed, :package:.

**Comment by barmac** [comment]
Amazing :)

**Pull request merged**